### ISSUE TITLE
Nipype not in GitHub dependency graphs

### ISSUE THREAD

**Issue by effigies** [nipy/nipype#2937]
It looks like GitHub is expanding their [dependencies](https://github.com/nipy/nipype/network/dependencies)/[dependents](https://github.com/nipy/nipype/network/dependents), which I suspect means that they consider their Python support good enough. If you've been watching this at all, you'll know that things that depend on nipype are linked to [satra/NiPypeold](https://github.com/satra/NiPypeold) (and nipy in general is poorly handled).

From talking with @vsoch (https://twitter.com/vsoch/status/1132477941168988160), I think the reason is that GitHub is parsing `setup.py`, and we do some fairly indirect stuff, there.

https://github.com/nipy/nipype/blob/460c1bb11d629d1a9a4455fb950cace49d176544/setup.py#L128-L130

In NiPypeold:

```Python
    setup( name = 'nipype',
           description = 'Nipype, a Neuroimaging in Python package',
```

This is also true of nibabel, which doesn't have any link:

<img width="718" alt="Screen Shot 2019-05-26 at 09 19 52" src="https://user-images.githubusercontent.com/83442/58382380-7159a380-7f97-11e9-89bf-d4651a19292b.png">

I think there are two questions:

1) Do we care about these graphs? They might be useful for grants, but they're pretty fragile... For instance, GitHub doesn't detect fMRIPrep's dependencies because we don't use `requirements.txt` except for when we need to pin to unreleased versions of dependencies.
2) Is there any advantage to the `info.py` approach to defining all setup.py attributes? I assume the idea was that a single `setup.py` could be copied to any package with maybe one line change to indicate where `info.py` should be found, but there's plenty of customization throughout these packages, as they all adapt to problems during different phases of distutils, setuptools, and the various packaging PEPs that have popped up.

The simple thing would be to include `name='nipype'` in `setup.py`, leave everything else alone, and see what happens.

**Comment by vsoch** [comment]
I don't think there is enough information to make any kind of decisions to change the code here based on our best guesses. The metrics are fairly new, likely to change, and it's not even clear they are correct (so grants likely wouldn't be impacted). If you really think this is meaningful information for grants, there is also nothing wrong with saying "The combined used by counts for nipype from (this date) to (this date) is N. 

TLDR: too early to be worried, too uncertain to be worried, and too trivial a metric to be worried. For now I think focus might be put on just figuring out how they calculate the metrics, and seeing if there is anything to be done (aside from chopping up perfectly good code) to make it more correct.

Now, if GitHub starts giving out buckets of moolahs based on this metric, and the repo is somehow important? Maybe then it would be time to care :)

**Comment by satra** [comment]
i agree with @vsoch - if anything github needs to improve it's tracking capabilities. it's also possible that it's doing super duper introspection and finds that nipypeold is the seed of everything!! (very unlikely though).

i think the better metrics for us is not just code, but all the people who are using it. @mgxd has been working on a telemetry server. perhaps we can get that up and running during the brainhack.

**Comment by effigies** [comment]
Sounds good. I've kind of looked at it a couple times and keep expecting Python support to improve, but was interpreting this new roll-out as a marker of what they think is stability.
